### PR TITLE
SA-587 Adding OpenAPI support

### PR DESCRIPTION
**Pull request by atomfinger**
## Description

Adding support for OpenAPI. First step is to add the new `web-spec` package and the correct configuraiton.

I'm doing this as part of a separate PR, as that will help separate configuration and implementation :)

### CODE REVIEW

**Comment by atomfinger** [comment]
Current dependencies on/for this PR:
* main
  * **PR #167** <a href="https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/167" target="_blank">Graphite</a>  👈
    * **PR #168** <a href="https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/168" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/protectorinsurance/protector-initializr-java/167?utm_source=stack-comment).

**Review by atomfinger — COMMENTED** [review]
ℹ  There are some files that has line endings changed. I'm not sure why. I'll look into it, but my IDE reports them as CRLF, so it should be fine.

**Review by atomfinger — COMMENTED** [review]
ℹ FYI: I'm testing out a tool called Graphite: graphite.dev

It is a tool to reduce the size of pull requests by stacking them. It is still something I'm experimenting with, but that is why you're seeing stacked PRs and branches. 

The theory is that I'll get better feedback with smaller PRs, and smaller PRs (albeit more) will increase the chance that a reviewer might just pick one up. There's no need for a single reviewer to verify each PR. 

The idea is that each PR should be small enough that it shouldn't be a huge time investment to review them. So maybe one reviews the first PR, maybe another person reviews the other and so forth.

File: web-schema/build.gradle [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                     useTags            : "true",
+                     skipDefaultInterface: "true"]
+    additionalProperties = [removeEnumValuePrefix: "false"]
+}
+
+task generateOpenApiSpec(type: GenerateTask) {
+    dependsOn(":web-schema:openApiGenerate")
+
+    generatorName = "openapi-yaml"
+    inputSpec = "$projectDir/../web-schema/src/main/resources/service.yml"
```
Comment (krgu99) [inline web-schema/build.gradle:33]: Burde vi ha tatt en rask diskusjon på YAML vs. JSON før vi velger en av de i initializer ? Personlig har jeg aldri likt YAML noe særlig, og tror det er en god del andre som deler denne oppfatningen.

**Review by krgu99 — APPROVED** [review]

File: web-schema/build.gradle (hunk 1 above)
Comment (atomfinger) [reply web-schema/build.gradle:33]: Jeg er ganske likegyldig, men føler at YAML er hakket ryddigere. Men det er bare min mening, og den er ikke så sterk akkurat her :)

Det jeg tenker er at vi kan kjøre en poll når folk kommer tilbake fra ferie og se hva flertallet ønsker. Mindretallet vil fortsatt lett kunne konvertere tilbake itl yaml om de skulle ønske det (2 min jobb).

Høres det greit ut?

Comment (krgu99) [reply web-schema/build.gradle:33]: Yes, det høres fint ut :)

File: web-schema/build.gradle [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+openApiGenerate {
+    generatorName = "spring"
+    inputSpec = "$projectDir/src/main/resources/service.yml"
+    outputDir = "$buildDir/generated"
+    apiPackage = "no.protector.insurable.cluster.web.openapi"
+    invokerPackage = "no.protector.insurable.cluster.web.openapi"
+    modelPackage = "no.protector.insurable.cluster.web.openapi.model"
+    modelNameSuffix = "Dto"
+    configOptions = [dateLibrary        : "java8",
```
Comment (pfs-olwe99) [inline web-schema/build.gradle:22]: Behöver dateLibrary specifieras explicit? java8 är default och borde aldrig ändras av oss.

File: web-schema/build.gradle [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    inputSpec = "$projectDir/../web-schema/src/main/resources/service.yml"
+    outputDir = "$buildDir/generated"
+}
+
+processResources.dependsOn tasks.generateOpenApiSpec
+processResources.dependsOn tasks.openApiGenerate
+compileJava.dependsOn tasks.openApiGenerate
+compileJava.dependsOn tasks.generateOpenApiSpec
+sourceSets.main.java.srcDir "$buildDir/generated/src/main/java"
+sourceSets.main.resources.srcDir "$buildDir/generated/src/main/resources"
```
Comment (pfs-olwe99) [inline web-schema/build.gradle:42]: Om möjligt, se till att det är newline på slutet av alla filer

Comment (atomfinger) [reply web-schema/build.gradle:42]: Gjør dette noe praktisk forskjell?

File: web-schema/build.gradle (hunk 2 above)
Comment (atomfinger) [reply web-schema/build.gradle:22]: Stemmer, jeg kan fikse det

**Pull request merged**